pandas/core/algorithms.py

```diff
@@ -948,10 +948,35 @@ def select_n_frame(frame, columns, n, method, keep):
     if not is_list_like(columns):
         columns = [columns]
     columns = list(columns)
-    ser = getattr(frame[columns[0]], method)(n, keep=keep)
-    if isinstance(ser, Series):
-        ser = ser.to_frame()
-    return ser.merge(frame, on=columns[0], left_index=True)[frame.columns]
+    ascending = method == 'nsmallest'
+    original_frame, original_index = frame, frame.index
+    frame.reset_index(drop=True, inplace=True)
```

Comment:
pls don't use ``inplace`` is completely non-idiomatic (not to mention doesn't offer *any* perf benefits).

``frame = frame.reset_index(drop=True)``